hazelcast/src/main/java/com/hazelcast/query/impl/predicates/InPredicate.java

```diff
@@ -68,11 +69,23 @@ public Predicate accept(Visitor visitor, Indexes indexes) {
 
     @Override
     protected boolean applyForSingleAttributeValue(Comparable attributeValue) {
-        if (attributeValue == null) {
+        Set<Comparable> set = convertedInValues;
+
+        if (attributeValue == null && set == null) {
```

Comment:
What if an `IN (null)` predicate is applied to a sequences of attribute values, where the first one is non-null and the second is null? The first call will result in initialising `set` which will cause the second iteration to use `set.contains()` - shouldn't it use `isNull()` for null items in the set?

Reply:
Good point, fixing...

Reply:
As it turned out, `isNull` is not really required here, `== null` is enough. `AbstractIndex.NULL` is produced while downgrading from index-based evaluation, see `CompositeEqualPredicate.apply` for instance, but there is no such downgrading for `InPredicate`. But still I decided to keep it for consistency with other predicates and added null normalization: https://github.com/hazelcast/hazelcast/pull/16568/commits/8c9f5a38e71587f5dcf1d1bdd0779932da92e753